htdocs/api/class/api_documents.class.php

```diff
@@ -425,7 +425,7 @@ public function getDocumentsListByElement($modulepart, $id = 0, $ref = '', $sort
 				throw new RestException(500, 'Error while fetching object: '.$object->error);
 			}
 
-			$upload_dir = $conf->product->multidir_output[$object->entity].'/'.get_exdir(0, 0, 0, 0, $object, 'product').dol_sanitizeFileName($object->ref);
+			$upload_dir = $conf->product->multidir_output[$object->entity]."/".dol_sanitizeFileName($object->ref);
```

Comment:
All getting of path must use the get_exdir (to prepare the future).
Can you check if this fix the trouble:
`upload_dir = $conf->product->multidir_output[$object->entity].'/'.get_exdir(0, 0, 0, 0, $object, 'product')`

Reply:
it seems to be better i get this but different of other objects. don't really easy for integration of this...

{
  "0": {
    "name": "PT2012-0002-xxx_squarred.jpg",
    "path": "/volume2/web/dolibarr/documents/produit/PT2012-0002",
    "level1name": "PT2012-0002",
    "relativename": "PT2012-0002-xxx_squarred.jpg",
    "fullname": "/volume2/web/dolibarr/documents/produit/PT2012-0002/PT2012-0002-xxx_squarred.jpg",
    "date": 1611946023,
    "size": 3988,
    "type": "file"
  },
  "ecmfiles_infos": [
    {
      "label": "96549a7e68ebfbb2b2cc32764bcaac4b",
      "entity": "1",
      "filename": "PT2012-0002-xxx_squarred.jpg",
      "filepath": "produit/PT2012-0002",
      "fullpath_orig": "xxx_squarred.jpg",
      "description": "",
      "keywords": null,
      "cover": null,
      "position": "2",
      "gen_or_uploaded": "uploaded",
      "extraparams": null,
      "date_c": 1611949623,
      "date_m": 1611949623,
      "fk_user_c": "1",
      "fk_user_m": null,
      "acl": null,
      "src_object_type": "product",
      "src_object_id": "2",
      "id": "45",
      "ref": null,
      "share": null
    }
  ]
}